msal/src/main/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplication.kt

```diff
@@ -333,184 +326,193 @@ class NativeAuthPublicClientApplication(
             methodName = "${TAG}.signIn(username: String, password: CharArray?, scopes: List<String>?)"
         )
         return withContext(Dispatchers.IO) {
-            try {
-
-                verifyNoUserIsSignedIn()
-
-                if (username.isBlank()) {
-                    return@withContext SignInError(
-                        errorType = ErrorTypes.INVALID_USERNAME,
-                        errorMessage = "Empty or blank username",
-                        correlationId = "UNSET"
-                    )
-                }
-
-                val hasPassword = password?.isNotEmpty() == true
-
-                val params =
-                    CommandParametersAdapter.createSignInStartCommandParameters(
-                        nativeAuthConfig,
-                        nativeAuthConfig.oAuth2TokenCache,
-                        username,
-                        password,
-                        scopes
-                    )
-
-                val command = SignInStartCommand(
-                    params,
-                    NativeAuthMsalController(),
-                    PublicApiId.NATIVE_AUTH_SIGN_IN_WITH_EMAIL
+            SignInResult.MFARequired(
+                nextState = AwaitingMFAState(
+                    continuationToken = "1234",
+                    correlationId = "abcd",
+                    scopes = scopes,
+                    config = nativeAuthConfig
                 )
+            )
```

Comment:
I'm confused by the logic here - won't this break all our sign in flows? Why are we removing all of the handling logic and hard-coding an MFA required response?

Reply:
The alternative is that the interface and business logic branches are lumped together into 1 PR, but that makes reviewing quite substantial

Reply:
Good suggestion, I updated this.